### PR TITLE
ci: enable new ci and build.yml changes

### PR DESCRIPTION
**Pull request by dsharma-dc**
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:

**What this PR does?**:

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [x] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: .github/workflows/build_and_push.yml [hunk 1]
```diff
@@ -247,3 +191,14 @@ jobs:
...
+  release-chart:
+    runs-on: ubuntu-latest
+    needs: ["csi-driver"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Publish lvm localpv develop or prerelease helm chart
+        uses: stefanprodan/helm-gh-pages@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          charts_dir: ./deploy/helm
```
Comment (Abhinandan-Purkait) [inline .github/workflows/build_and_push.yml]: Extra line please?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: .github/workflows/build_and_push.yml (hunk 1 above)
Comment (dsharma-dc) [reply .github/workflows/build_and_push.yml]: done.
Also added crds deletion in test script's cleanup function.

File: ci/ci-test.sh [hunk 2]
```diff
@@ -41,10 +42,11 @@ cleanup() {
 
   cleanup_foreign_lvmvg
 
-  kubectl delete pvc -n openebs lvmpv-pvc
+  kubectl delete pvc -n "$OPENEBS_NAMESPACE" lvmpv-pvc
   kubectl delete -f "${SNAP_CLASS}"
 
   helm uninstall lvm-localpv -n "$OPENEBS_NAMESPACE" || true
+  kubectl delete crds $CRDS_TO_DELETE_ON_CLEANUP
```
[the remark this answers is not in the document]
Comment (dsharma-dc) [reply ci/ci-test.sh]: ok bot, done !

File: deploy/helm/charts/templates/lvm-controller.yaml [hunk 3]
```diff
@@ -109,7 +109,7 @@ spec:
               value: controller
             - name: OPENEBS_CSI_ENDPOINT
               value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
-            - name: LVM_NAMESPACE
+            - name: OPENEBS_NAMESPACE
```
Comment (tiagolobocastro) [inline deploy/helm/charts/templates/lvm-controller.yaml:112]: This doesn't seem like just a CI change, should be on it's own commit?
Does the code already support this?

**Review by tiagolobocastro — COMMENTED** [review]
How is tagged release handled?

File: deploy/helm/charts/templates/lvm-controller.yaml (hunk 3 above)
Comment (dsharma-dc) [reply deploy/helm/charts/templates/lvm-controller.yaml:112]: The code already supports this new name.

**Comment by dsharma-dc** [comment]
> How is tagged release handled?

That's in #340 if that's what you are asking.

File: deploy/helm/charts/templates/lvm-controller.yaml (hunk 3 above)
Comment (tiagolobocastro) [reply deploy/helm/charts/templates/lvm-controller.yaml:112]: Should still be in its own commit, this is not a CI change

Comment (dsharma-dc) [reply deploy/helm/charts/templates/lvm-controller.yaml:112]: Separated helm and CI change commits.

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsharma-dc** [comment]
@Abhinandan-Purkait I see a very strange thing, that the output of below command that we have in workflow files varies between zfs vs lvm repos.
> awk -F': ' '/^version:/ {print $$2}' deploy/helm/charts/Chart.yaml

In zfs, it correctly returns e.g. `2.7.0-develop`. In lvm it is returning the first field i.e. the string `version` ! I don't see any difference in Chart.yaml formattting. Using single `$` in `print` works correctly. I will change this to use single `$` as I think it's just a typo that there are two `$$` inadvertently.

**Pull request merged**